sim/gui.py

```diff
@@ -504,14 +507,44 @@ def main_loop(self):  # pylint: disable=too-many-locals
                 Position(0, 2 * self.tile_size + self.half_tile),
             )
             self.draw_text(
-                f'Goal(s): {self.sim.end}',
+                f'Steps: {self.sim.counter.current_step_counter}',
+                self.text_size,
+                Position(middle_width - 6 * self.tile_size, 2 * self.tile_size + self.half_tile),
+            )
+            self.draw_text(
+                f'Weight: {self.sim.counter.current_weight_counter}',
+                self.text_size,
+                Position(middle_width, 2 * self.tile_size + self.half_tile),
+            )
+            self.draw_text(
+                f'Cells: {self.sim.counter.current_cell_counter}',
+                self.text_size,
+                Position(middle_width + 7 * self.tile_size, 2 * self.tile_size + self.half_tile),
+            )
+            self.draw_text(
+                f'Goal(s): {str(self.sim.end).replace('frozenset', '')[2:-2]}',
```

Comment:
```suggestion
                f'Goal(s): {", ".join(map(str, self.sim.end))}',
```
or
```suggestion
                f'Goal(s): {", ".join(str(g) for g in self.sim.end)}',
```

(and because you wanted to make it a `list`)
```suggestion
                f'Goal(s): {", ".join(map(str, sorted(self.sim.end)))}',
```
or
```suggestion
                f'Goal(s): {", ".join(str(g) for g in sorted(self.sim.end))}',
```